DEVELOPER-GUIDE.md

```diff
@@ -0,0 +1,82 @@
+# 1-Click App Development with Packer
+
+This guide outlines the essential files and considerations for creating a 1-Click app deployment using Packer.
+
+## Basic 1-Click Setup
+
+**Required Files:**
+
+* `example-1-click/template.json`: This file defines the packages to install, image build configuration, and installation locations.
+* `example-1-click/scripts/installer.sh`: This script performs the actual application installation on the Droplet.
+* `example-1-click/files/etc/update-motd.d/99-one-click`: This file contains basic instructions for users upon first login to the Droplet.
+
+**Additional Considerations:**
+For a very basic 1-Click, above is all you need. However, there might be scenarios where:
+
+**Startup Script:**
+
+    - You need to run a script to setup something when the droplet starts rather than during the build process.
+    - `example-1-click/files/var/lib/cloud/scripts/per-instance/001_onboot`: This script executes commands during Droplet startup.
+
+**Systemd Services:**
+
+    - If you need some systemd services to be installed as part of the build process, you add them under `example-1-click/files/etc/systemd/system` . Notice that the path after `example-1-click/files` must be exactly how you want it in the droplet. Since we want the systemd services under `/etc/systemd/system` we did the same. You can then enable/start these services in the `installer.sh` or `001_onboot` .
+    - The directory structure within `example-1-click/files` mirrors the desired structure on the built Droplet (e.g., `/etc/systemd/system` for systemd services).
+
+**Application Configuration:**
+
+    - `example-1-click/files/etc/nginx/sites-available/`: This directory stores configuration files for services like Nginx (the path may vary depending on your application).
+    - Example: `sites.conf` for Nginx configuration, located at `/etc/nginx/sites-available/` on the Droplet.
+
+**Additional Files:**
+
+    - `example-1-click/files/var/lib/digitalocean/`: This directory holds any files referenced by scripts like `001_onboot`.
+
+**Packer Configuration (`template.json`):**
+
+```json
+  [
+    {
+      "type": "file",
+      "source": "example-1-click/files/etc/",
+      "destination": "/etc/"
+    },
+    {
+      "type": "file",
+      "source": "example-1-click/files/var/",
+      "destination": "/var/"
+    },
+    // ... Add more file copy blocks as needed
+    {
+      "type": "shell",
+      "environment_vars": [
+        "application_name={{user `application_name`}}",
+        "application_version={{user `application_version`}}",
+        "DEBIAN_FRONTEND=noninteractive",
+        "LC_ALL=C",
+        "LANG=en_US.UTF-8",
+        "LC_CTYPE=en_US.UTF-8"
+      ],
+      "scripts": [
+        "example-1-click/scripts/installer.sh"
```

Comment:
Do we want to include the recommended scripts here as well, since this is meant to be an example? Specifically the cleanup and force-ssh-logout ones, since those are used in just about all of these.

Reply:
Those are already defined in the section below

Reply:
Right. I'm thinking if someone copies this as an example to start from, though. It might be nice to have the recommended scripts included there.

Reply:
Updated